cmd/ipfs/init.go

```diff
@@ -102,31 +109,41 @@ environment variable:
 			}
 		}
 
-		profile, _ := req.Options[profileOptionName].(string)
+		cfgLocation, _ := req.Options[configOptionName].(string)
+		profiles, _ := req.Options[profileOptionName].(string)
 
-		var profiles []string
-		if profile != "" {
-			profiles = strings.Split(profile, ",")
+		if cfgLocation != "" {
+			conf, err := cserial.Load(cfgLocation)
+			if err != nil {
+				return err
+			}
+
+			return doInit(os.Stdout, cctx.ConfigRoot, false, nBitsForKeypairDefault, profiles, conf)
```

Comment:
Why not use the `doInit` call below? This one drops the `empty` flag and passes the wrong `nBitsForKeypair` option (although I guess _that_ doesn't do anything right now).